### PR TITLE
Yarn merge from upstream

### PR DESCRIPTION
**Pull request by tijsverkoyen**
## Type

- Enhancement
- Feature

## Pull request description

Install needed assets with [`yarn`](https://yarnpkg.com), and using `gulp` to generate the css based on the scss-files.

Yarn is a package manager which has a `lock`-file per default, so everybody should get the correct versions, this in favor of `npm`, which does not have a locking mechanism.

The `gulp build`-task will move the required assets into the correct folders and will regenerate the CSS-files for the backend.

At this moment the "moved" assets are included in the repo for two reasons: 

1. We don't have an automated way to include the assets when releasing
2. We want users to be able to install Fork CMS without any knowledge of Sass/Gulp/...

With this pull-request we should improve the disk-space usage by a fair bit, as the bower_components-folder is removed and only the required files are included.

Also we don't rely on `bower` anymore.

### CODE REVIEW

**Comment by tijsverkoyen** [comment]
Some remarks:

* `bower.json`
    * `modernizr` was required, but in the theme it is loaded from a CDN?
* old `gulpfile.js`
    * `gulp-clean` is deprecated in favor of `del`
    * `runSequence` shouldn't be used, as the required functionality is available with default gulp
    * Why do we only generate map-files in dev-mode?
* new `gulpfile.js`
    * I renamed most of the methods to be more descriptive
    * default should be build instead of watch in my opinion

File: package.json [hunk 1]
```diff
@@ -7,14 +7,39 @@
   "author": "Tijs Verkoyen <tijs@verkoyen.eu>",
   "license": "MIT",
   "devDependencies": {
+    "babel-core": "^6.21.0",
+    "babel-loader": "^6.2.10",
+    "babel-preset-es2015": "^6.18.0",
+    "babel-preset-react": "^6.16.0",
```
Comment (mathiashelin) [inline package.json]: I would drop React.

File: package.json [hunk 2]
```diff
@@ -7,14 +7,39 @@
...
+    "lodash": "^4.17.4",
+    "requestanimationframe": "^0.0.23",
+    "sweet-scroll": "^2.2.0",
+    "webpack": "^1.14.0",
+    "webpack-stream": "^3.2.0"
+  },
+  "babel": {
+    "presets": [
+      "es2015",
+      "react"
```
Comment (mathiashelin) [inline package.json]: Drop React

File: package.json (hunk 1 above)
Comment (tijsverkoyen) [inline package.json]: Fixed in d4f4c49593ec774b2f0f543e8a4bc72f0c9e6a6c

File: package.json (hunk 2 above)
Comment (tijsverkoyen) [inline package.json]: Fixed in d4f4c49593ec774b2f0f543e8a4bc72f0c9e6a6c

**Pull request merged**